client/js/bar.js

```diff
@@ -59,4 +54,9 @@ function createBarChart(chartValues, chartCategories) {
   }
 }
 
-createBarChart(CHART_VALUES, CHART_CATEGORIES);
+GENRE_ANALYSIS.then((DATA) => {
+  createBarChart(
+      Object.values(DATA.genreData),
+      Object.keys(DATA.genreData),
+      DATA.maxGenreCount);
```

Comment:
I'm not sure if `Object.values` and `Object.keys` is the best way to do it

I assume that the array returned from both methods should match up with each other

Reply:
If they match up then I think this is a clever-ish solution to your problem. You could also just change createBarChart to accept the entire map and then loop through the key-value pairs there. I personally prefer the second method, but I'd also like to see what @laptou has to say about it.